src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs

```diff
@@ -724,6 +724,55 @@ string GetSourceAndId()
             }
         }
 
+        private static object GetValueFromRolledUpEntry(int index, SingleResult singleResult)
+        {
+            // we are working with a rolled-up series
+            // here an entry has 6 different values (min, max, first, last, sum, count) per each 'original' measurement
+            // we need to return the average value (sum / count)
+
+            index *= 6;
+            if (index + (int)AggregationType.Count >= singleResult.Values.Length)
+                return null;
+            if (singleResult.Values.Span[index + (int)AggregationType.Count] == 0)
+                return double.NaN;
+            return singleResult.Values.Span[index + (int)AggregationType.Sum] / singleResult.Values.Span[index + (int)AggregationType.Count];
+        }
+
+        private object GetNamedValue(string fieldValue, SingleResult singleResult, bool isRaw)
+        {
+            if (_databaseRecord == null)
+            {
+                var dbId = Constants.Documents.Prefix + _context.DocumentDatabase.Name;
+                using (_context.DocumentDatabase.ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext txContext))
+                using (txContext.OpenReadTransaction())
+                using (var dbDoc = _context.DocumentDatabase.ServerStore.Cluster.Read(txContext, dbId, out long etag))
+                {
+                    _databaseRecord = DocumentConventions.Default.Serialization.DefaultConverter.FromBlittable<DatabaseRecordWithEtag>(dbDoc, "database/record");
```

Comment:
use `ReadRawDatabaseRecord` and then `TimeSeriesConfiguration`,  we don't need the entire database record here